### ISSUE TITLE
Multiline tuples in type declarations are not allowed

### ISSUE THREAD

**Issue by edkv** [avh4/elm-format#415]
When using type declarations with tuples like this
```
type alias SomeType =
    ( Int -> Int -> Int
    , String -> String -> String
    )


type AnotherType =
    AnotherType
        ( Int -> Int -> Int
        , String -> String -> String
        )
```

`elm-format` reformats it as
```
type alias SomeType =
    ( Int -> Int -> Int, String -> String -> String )


type AnotherType
    = AnotherType ( Int -> Int -> Int, String -> String -> String )
```

This behaviour hurts readability and may create very long lines.

**Comment by avh4** [comment]
Can you share any details about the real code where you encounter this?  (The general recommendation is to switch to using records when you start having complicated data structures.)

**Comment by edkv** [comment]
Well, what I'm doing already goes far beyond general recommendations :) I'm experimenting with an implementation of components in elm. It looks like this:

```elm
div []
    [ counter
        ( .firstCounter
        , \v c -> { c | firstCounter = v }
        )
    , counter
        ( .secondCounter
        , \v c -> { c | secondCounter = v }
        )
    ]
```
As you can see, the data structure isn't really complicated, it's just a pair of functions bundled together that are used to convert children to parent types.

Earlier it was a record:
```elm
counter
    { get = .firstCounter
    , set = \v c -> { c | firstCounter = v }
    }
```
But I switched to tuples as it's shorter and seems more readable to me. When I created the issue, the type for that tuple was

```elm
type alias Slot children msg state pChildren =
    ( pChildren -> Container children msg state, Container children msg state -> pChildren -> pChildren )
```
Now it's simpler:
```elm
type alias Slot container pChildren =
    ( pChildren -> container, container -> pChildren -> pChildren )
```
So it's not that bad now, but I still would like to split it in two lines because that would be more readable in my opinion.

Also, notice how I used multiline tuples in the example with counters above. I also write it like that in updates quite often for readability purposes
```elm
update Msg -> Model -> ( Model, Cmd Msg )
update msg model =
    case msg of
        UpdateValue value ->
             ( { model | value = value }
             , doSideEffectWithValue value
             )
```
And `elm-format` allows it just fine, so I expected to be able to use similar approach in type declarations as well.

**Comment by avh4** [comment]
Should be fixed in 0.8.4 via https://github.com/avh4/elm-format/pull/683

**Comment by avh4** [comment]
0.8.4 is now released with this fix: https://github.com/avh4/elm-format/releases/tag/0.8.4
